content/en/apps/guides/integrations/best-practices/rapidpro.md

```diff
@@ -0,0 +1,108 @@
+- Customize messages as much as possible by pre-populating information already known such as name, to target respondents, especially when phones are shared among clients.
+- Translate all messages, especially when deploying in a multi-lingual environment. This ensures that respondents fully understand the survey in their language.
+- Beware of the timing of the surveys that directly affects response rates. From experience, sending questions when respondents are busy with their errands during the day ultimately leads to low response rates as opposed to evenings when they are done for the day.
+- Make sure you handle unsolicited responses by redirecting such to for example, a flow that eventually alerts concerned individuals e.g. reports of an outbreak etc.
+
+### Configuration
+1. Persist a unique identifier for each respondent entering the flows. This identifier shall be used to link flow runs to the recipients to make data analysis smooth.
+2. Use consistent language and design patterns.
+3. Be mindful of when you save data back to the CHT. From experience, this should happen at major milestones in a flow e.g. end of a flow or before sending a payload to an API endpoint.
+4. Utilize `globals`, shared values that can be referenced in flows, as well as broadcasts and campaigns, within your account referenced by `@globals.value_name`. This avoids re-entry of values in various components.
+5. Beware of concurrency that is not supported in RapidPro. Concurrency refers to a situation whereby one contact participates in more than one flows at the same time. Whenever this happens, the former flow shall be interrupted in favor of the latter. This can result in respondents exiting flows before completion, which impacts data quality. A master flow that controls individual flows is recommended. 
+6. Persist a respondent’s state in CHT. A state document is a JSON document that has calculated values of a respondent’s attributes or ledger of reports that is used by RapidPro to check the flows or questions one is eligible to receive. This document should not be editable by users directly. A sample is as follows:
+```json
+{
+  "patient_uuid": "0123...",
+  "language": "en",
+  "tb": "true",
+  "covid": "mild",
+  ...
+}
+```
+
+### Testing
+Testing includes manual and scripted.
+- Manual
+   - Test the workflows as close to production as possible. Telegram is a convenient way of validating the flows, but might not give a true picture of SMS in case of SMS messaging. Such testing will check SMS fidelity too.
+   - Piloting prior to a rollout is recommended. Remember, this shall expose the flows to real respondents and be helpful towards uncovering details such as text display among others.
+   - In order to prevent breakages in flows test end-to-end and fully, especially when edits and changes are introduced to check that the entire flow is not impacted by the change. 
```

Comment:
```suggestion
   - In order to prevent breakages in flows test end-to-end and fully as even minor edits/changes can have unpredictable impacts. Check that the entire flow is not impacted by the change prior to releasing in production.
```